src/Language/DifferentialDatalog/Compile.hs

```diff
@@ -2592,6 +2596,15 @@ mkExpr' d ctx EAs{..} | bothIntegers && narrow_from && narrow_to && width_cmp /=
 
 mkExpr' _ _ e = error $ "Compile.mkExpr': unexpected expression at " ++ show (pos e)
 
+mkFuncName :: DatalogProgram -> Function -> Doc
+mkFuncName d f | length namesakes == 1 = rname $ name f
+               | otherwise =
+    (rname $ name f) <> "_" <> targ0 <> "_" <> pp (length $ funcArgs f)
+    where
+    arg0 = funcArgs f !! 0
```

Comment:
why can't this go out of bounds?

Reply:
`arg0` is only evaluated if this is a polymorphic function.  Polymorphic functions must take at least one argument (enforced in Validate.hs)